cloudmanager/projects_test.go

```diff
@@ -0,0 +1,384 @@
+package cloudmanager
+
+import (
+	"fmt"
+	"github.com/mongodb/go-client-mongodb-atlas/mongodbatlas"
+	"net/http"
+	"reflect"
+	"testing"
+
+	"github.com/go-test/deep"
+)
+
+func TestProject_GetAllProjects(t *testing.T) {
+	setup()
+	defer teardown()
+
+	mux.HandleFunc("/groups", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, http.MethodGet)
+		_, _ = fmt.Fprint(w, `{
+			"links": [{
+				"href": "https://cloud.mongodb.com/api/public/v1.0/groups",
+				"rel": "self"
+			}],
+			"results": [{
+				"activeAgentCount": 0,
+				"hostCounts": {
+					"arbiter": 0,	
+					"config": 0,
+					"master": 0,
+					"mongos": 0,
+					"primary": 0,
+					"secondary": 0,
+					"slave": 0
+				},
+				"id": "56a10a80e4b0fd3b9a9bb0c2",
+				"lastActiveAgent": "2016-03-09T18:19:37Z",
+				"links": [{
+					"href": "https://cloud.mongodb.com/api/public/v1.0/groups/56a10a80e4b0fd3b9a9bb0c2",
+					"rel": "self"
+				}],
+				"name": "012i3091203jioawjioej",
+				"orgId": "5980cfdf0b6d97029d82f86e",
+				"publicApiEnabled": true,
+				"replicaSetCount": 0,
+				"shardCount": 0,
+				"tags": []
+			}, {
+				"activeAgentCount": 0,
+				"hostCounts": {
+					"arbiter": 0,
+					"config": 0,
+					"master": 0,
+					"mongos": 0,
+					"primary": 0,
+					"secondary": 0,
+					"slave": 0
+				},
+				"id": "56aa691ce4b0a0e8c4be51f7",
+				"lastActiveAgent": "2016-01-29T19:02:56Z",
+				"links": [{
+					"href": "https://cloud.mongodb.com/api/public/v1.0/groups/56aa691ce4b0a0e8c4be51f7",
+					"rel": "self"
+				}],
+				"name": "1454008603036",
+				"orgId": "5980d0040b6d97029d831798",
+				"publicApiEnabled": true,
+				"replicaSetCount": 0,
+				"shardCount": 0,
+				"tags": []
+			}],
+			"totalCount": 2
+		}`)
+	})
+
+	projects, _, err := client.Projects.GetAllProjects(ctx)
+	if err != nil {
+		t.Errorf("Projects.GetAllProjects returned error: %v", err)
+	}
+
+	expected := &Projects{
```

Comment:
nit: it would probably be easier to define this in a json (test) resource and just deserialize it in the test.